### DISCUSSION TITLE
App verification and vapid_key usecase

### DISCUSSION THREAD

**Discussion by sudhagar-mr-17051** [mastodon/mastodon#24330] [Q&A] [answered]
After creating app using /api/v1/apps ,

is it mandatory to check whether the app is working properly using /api/v1/apps/verify_credentials endpoint ??

vapid_key is used for what purpose?, we are building a web application and integrating mastodon in it, should we need to store vapid_key for any purpose?

**Comment by andypiper** [comment, marked as answer]
The `/verify_credentials` endpoint is typically for confirming that you have a valid set of OAuth credentials - it is not mandatory, but it's the first thing to check to confirm that you have a valid token before making other calls.

The `vapid_key` is using for the Push Streaming API, and is documented [here](https://docs.joinmastodon.org/entities/Application/#vapid_key).

**Reply by sudhagar-mr-17051** [reply]
Thanks for the clarification
